Modules/Loadable/VolumeRendering/MRML/vtkMRMLVolumeRenderingDisplayNode.h

```diff
@@ -157,6 +162,8 @@ class VTK_SLICER_VOLUMERENDERING_MODULE_MRML_EXPORT vtkMRMLVolumeRenderingDispla
 
   int CroppingEnabled;
 
+  int ROIDisplayEnabled;
```

Comment:
I really appreciate your work that you've put into this, but unfortunately this solution is not very clean. The issue is that ROI visibility is already stored in the scene (for example, you can access it by getNode('AnnotationROI').GetDisplayVisibility()). If we add a variable that stores this status then we have to implement a synchronization mechanism, which is an unnecessary complexity. Instead, the GUI should just observe the displayable node vtkMRMLDisplayableNode::DisplayModifiedEvent and update the GUI if the event is invoked.